sjb/config/test_cases/test_branch_origin_extended_conformance_install_openstack.yml

```diff
@@ -0,0 +1,193 @@
+    title: "exit if no capacity"
+    repository: "openshift-ansible"
+    script: |-
+      if hash openstack &>/dev/null; then
+          stack_count="$(openstack stack list -f value | grep -cv ${buildId}.local || true)"
+      else
+          stack_count=0
+      fi
+      if [ "$stack_count" -ge "${CI_CONCURRENT_JOBS}" ]; then
+          echo "Openstack provider CI is over capacity, skipping the end-end test."
+          exit 1
+      fi
+  - type: "script"
+    title: "setup teardown hook"
+    repository: "openshift-ansible"
+    script: |-
+      teardown(){
+          openstack keypair delete "${buildId}" || true
+          openstack stack delete --wait --yes "${buildId}.local" || true
+      }
+      trap teardown EXIT
+  - type: "script"
+    title: "install origin on openstack"
+    repository: "openshift-ansible"
+    script: |-
+      access_ip="$(curl --silent https://api.ipify.org)"
+      openstack keypair create "${buildId}" > ~/.ssh/id_rsa
+      chmod 600 ~/.ssh/id_rsa
+
+      cat > ansible.cfg << EOF_CAT
+      [defaults]
+      ansible_user = openshift
+      forks = 50
+      timeout = 180
+      host_key_checking = false
+      inventory = playbooks/openstack/sample-inventory
+      inventory_ignore_extensions = secrets.py, .pyc, .cfg, .crt
+      gathering = smart
+      retry_files_enabled = false
+      fact_caching = jsonfile
+      fact_caching_connection = .ansible/cached_facts
+      fact_caching_timeout = 900
+      stdout_callback = skippy
+      callback_whitelist = profile_tasks
+      [ssh_connection]
+      ssh_args = -o ControlMaster=auto -o ControlPersist=900s -o GSSAPIAuthentication=no
+      control_path = /var/tmp/%%h-%%r
+      pipelining = True
+      EOF_CAT
+
+      cat > extra-vars.yaml << EOF_CAT
+      openshift_master_unsupported_embedded_etcd: True
+      openshift_repos_enable_testing: True
+      openshift_openstack_ephemeral_volumes: True
+      rhsm_register: False
+      openshift_openstack_num_masters: 1
+      openshift_openstack_num_infra: 1
+      openshift_openstack_num_nodes: 1
+      openshift_openstack_num_etcd: 0
+      openshift_openstack_external_network_name: $EXT_NET_NAME
+      openshift_openstack_default_image_name: $IMAGE_NAME
+      openshift_openstack_node_ingress_cidr: "${access_ip}/32"
+      openshift_openstack_ssh_ingress_cidr: "${access_ip}/32"
+      openshift_openstack_lb_ingress_cidr: "${access_ip}/32"
+      openshift_openstack_keypair_name: $buildId
+      openshift_openstack_clusterid: $buildId
+      openshift_openstack_public_dns_domain: local
+      openshift_openstack_dns_nameservers:
+        - 208.67.222.220
+        - 38.145.32.66
+      openshift_master_identity_providers:
+      - name: 'htpasswd_auth'
+        login: 'true'
+        challenge: 'true'
+        kind: 'HTPasswdPasswordIdentityProvider'
+        filename: '/etc/origin/master/htpasswd'
+      openshift_master_htpasswd_users:
+        test: '\$apr1\$vUfm7jQS\$C6Vn0GDScgOjzvk1PSHe1/'
```

Comment:
note : this is just a throw-away thing, it's ephemeral for a job live time, and access to public IP is restricted to only the worker node public IP